### PR TITLE
performance: memoize a few things

### PR DESCRIPTION
**Pull request by adrianthedev**
# Description

Trying out a few performance improvements.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: app/controllers/avo/base_controller.rb [hunk 1]
```diff
@@ -103,7 +103,7 @@ def new
 
       # Handle special cases when creating a new record via a belongs_to relationship
       if params[:via_belongs_to_resource_class].present?
-        return render turbo_stream: turbo_stream.append('attach_modal', partial: 'avo/base/new_via_belongs_to')
+        return render turbo_stream: turbo_stream.append("attach_modal", partial: "avo/base/new_via_belongs_to")
```
Comment (adrianthedev) [inline app/controllers/avo/base_controller.rb:106]: Lint.

File: app/controllers/avo/base_controller.rb [hunk 2]
```diff
@@ -149,7 +149,7 @@ def create
 
           if params[:via_association_type] == "has_one"
             # On has_one scenarios we should switch the @record and @related_record
-            @related_record.send("#{@reflection.parent_reflection.inverse_of.name}=", @record)
+            @related_record.send(:"#{@reflection.parent_reflection.inverse_of.name}=", @record)
```
Comment (adrianthedev) [inline app/controllers/avo/base_controller.rb:152]: Lint.

File: app/components/avo/resource_component.rb [hunk 3]
```diff
@@ -102,7 +103,7 @@ def has_reflection_and_is_read_only
   end
 
   def render_control(control)
-    send "render_#{control.type}", control
+    send :"render_#{control.type}", control
```
Comment (adrianthedev) [inline app/components/avo/resource_component.rb:106]: Lint.

File: app/controllers/avo/base_controller.rb [hunk 4]
```diff
@@ -442,7 +442,7 @@ def create_success_action
       return render "close_modal_and_reload_field" if params[:via_belongs_to_resource_class].present?
 
       respond_to do |format|
-        format.html { redirect_to after_create_path, notice: create_success_message}
+        format.html { redirect_to after_create_path, notice: create_success_message }
```
Comment (adrianthedev) [inline app/controllers/avo/base_controller.rb:445]: Lint.

File: app/controllers/avo/base_controller.rb [hunk 5]
```diff
@@ -368,7 +368,7 @@ def set_actions
     def set_applied_filters
       reset_filters if params[:reset_filter]
 
-      # Return if there are no filters or if the filters are actualy ActionController::Parameters (used by dynamic filters)
+      # Return if there are no filters or if the filters are actually ActionController::Parameters (used by dynamic filters)
```
Comment (adrianthedev) [inline app/controllers/avo/base_controller.rb:371]: typo

File: app/controllers/avo/base_controller.rb [hunk 6]
```diff
@@ -583,7 +583,7 @@ def set_component_for(view, fallback_view: nil)
       ).handle
 
       # If the component is not set, use the default one
-      if (custom_component = components.dig("resource_#{view}_component".to_sym)).nil?
+      if (custom_component = components.dig(:"resource_#{view}_component")).nil?
```
Comment (adrianthedev) [inline app/controllers/avo/base_controller.rb:586]: Lint.

File: lib/avo/concerns/hydration.rb [hunk 7]
```diff
@@ -1,13 +1,11 @@
...
   module Concerns
     module Hydration
-      extend ActiveSupport::Concern
-
       def hydrate(**args)
         args.each do |key, value|
           value = Avo::ViewInquirer.new value if key == :view
 
-          send("#{key}=", value) if respond_to?("#{key}=")
+          send(:"#{key}=", value) if respond_to?(:"#{key}=")
```
Comment (adrianthedev) [inline lib/avo/concerns/hydration.rb:8]: Lint.

**Review by Paul-Bob — APPROVED** [review]

**Pull request merged**